From: Li Wang <liwang@redhat.com>
Subject: selftests/lib.mk: set PWD from CURDIR to avoid wrong extmod path
Date: Tue, 31 Mar 2026 12:01:55 +0800

When running selftests from the kernel top-level (e.g. make kselftest-all),
sub-makes might still inherit the caller's PWD from the environment.

Some selftests use $(PWD) in recursive kbuild invocations, which can
then incorrectly resolve to the kernel top directory instead of the
current test directory.

In that case, kbuild may generate an external-module wrapper Makefile in
the wrong location, potentially clobbering the top-level Makefile and
causing recursive include failures ("Too many open files").

Export PWD := $(CURDIR) in selftests/lib.mk so $(PWD) always matches the
actual current selftest directory.

Link: https://lkml.kernel.org/r/20260331040156.119158-3-liwang@redhat.com
Signed-off-by: Li Wang <liwang@redhat.com>
Reported-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/lib.mk |    2 ++
 1 file changed, 2 insertions(+)

--- a/tools/testing/selftests/lib.mk~selftests-libmk-set-pwd-from-curdir-to-avoid-wrong-extmod-path
+++ a/tools/testing/selftests/lib.mk
@@ -60,6 +60,8 @@ endif
 selfdir = $(realpath $(dir $(filter %/lib.mk,$(MAKEFILE_LIST))))
 top_srcdir = $(selfdir)/../../..
 
+export PWD := $(CURDIR)
+
 # msg: emit succinct information message describing current building step
 # $1 - generic step name (e.g., CC, LINK, etc);
 # $2 - optional "flavor" specifier; if provided, will be emitted as [flavor];
diff --git a/tools/testing/selftests/mm/local_config.stamp a/tools/testing/selftests/mm/local_config.stamp
new file mode 100644
_
